include/spdlog/sinks/qt_sinks.h

```diff
@@ -4,46 +4,55 @@
 #pragma once
 
 //
-// Custom sink for QPlainTextEdit or QTextEdit and its childs(QTextBrowser...
-// etc) Building and using requires Qt library.
+// Custom sink for QPlainTextEdit or QTextEdit and its children(QTextBrowser...
+// etc) also any QObject will be ok with QString slot. Building and using requires Qt library.
 //
 
 #include "spdlog/common.h"
 #include "spdlog/details/log_msg.h"
 #include "spdlog/details/synchronous_factory.h"
 #include "spdlog/sinks/base_sink.h"
 
-#include <QTextEdit>
-#include <QPlainTextEdit>
+#if defined(SPDLOG_FMT_EXTERNAL)
+#include <fmt/color.h>
+#elif !defined(SPDLOG_USE_STD_FORMAT)
+#include "spdlog/fmt/bundled/color.h"
+#endif
+
+#include <QObject>
+
+class QTextEdit;
+class QPlainTextEdit;
 
 //
 // qt_sink class
 //
 namespace spdlog {
 namespace sinks {
-template<typename Mutex>
-class qt_sink : public base_sink<Mutex>
-{
+#if !defined(SPDLOG_USE_STD_FORMAT)
+template <typename Mutex>
+class qt_sink_color;
+#endif
+template <typename Mutex>
+class qt_sink : public base_sink<Mutex> {
+#if !defined(SPDLOG_USE_STD_FORMAT)
+    friend class qt_sink_color<Mutex>;
```

Comment:
why friend not needed?

Reply:
I want to access qt_sink's priv members from qt_sink_color class